### PR TITLE
Integration specs

### PR DESCRIPTION
**Pull request by evillaz**
This branch's purpose was to fulfill:
- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.
  -In your PR description include the screenshots of your app's logs before and after the fix
before-after

- User index page:

  - I can see the username of all other users.
  - I can see the profile picture for each user.
  - I can see the number of posts each user has written.
  - When I click on a user, I am redirected to that user's show page.

- User show page:

  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see the user's bio.
  - I can see the user's 3 recent posts.
  - I can see a button that lets me view all of a user's posts.
  - When I click a user's post, it redirects me to that post's show page.
  - When I click to see all posts, it redirects me to the user's post's index page.

- User post index page:

  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see a post's title.
  - I can see some of the post's body.
  - I can see the recent comments on a post.
  - I can see how many comments a post has.
  - I can see how many likes a post has.
  - I can see a section for pagination if there are more posts than fit on the view.
  - When I click on a post, it redirects me to that post's show page.

- Post show page:

  - I can see the post's title.
  - I can see who wrote the post.
  - I can see how many comments it has.
  - I can see how many likes it has.
  - I can see the post body.
  - I can see the username of each commentor.
  - I can see the comment each commentor left.
  
  There are some tests that imply using the first posts or comments. I changed them to the "recent" posts and comments instead, to use the method we created on the first sessions. "recent_posts" and "recent_comments"

### CODE REVIEW

File: spec/integration/user_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  context 'when viewing the page' do
+    it 'shows the profile picture for user.' do
+      expect(page).to have_css("img[src*='#{user.photo}']")
+    end
+    it 'shows the users username.' do
+      expect(page).to have_content(user.name)
+    end
+    it 'shows how many posts each user has written.' do
+      expect(page).to have_content(user.posts_counter)
+    end
```
Comment (GzimAsani) [inline spec/integration/user_show_spec.rb:25]: - [ ] You have a failing test, kindly fix it: 
image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello, @evillaz and @Kaghenimbale                        :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job guys, you did great with test coverage :+1: 

 ## Required Changes 

- [ ] Check for the comments below :arrow_double_down: 

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N/A



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️

Hi @evillaz , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @evillaz and @Kaghenimbale, 

# Required Changes ♻️

so-close-jim-halpert

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

- Use Capybara ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- No linter errors ✔️ 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [ ] All tests was executed successfully, but there are three tests that didn't pass, please fix your tests to pass them all and approve this milestone 👇🏽 

```ruby
franclobo@DESKTOP-E09US2D:~/microverse/blog-app-3$ rspec spec .
**Capybara starting Puma...
* Version 5.6.6 , codename: Birdie's Version
* Min threads: 0, max threads: 4
* Listening on http://127.0.0.1:43601
.......................................F........F.....F...

Pending: (Failures listed here are expected and do not affect your suite's status)

  1) PostsHelper add some examples to (or delete) /home/franclobo/microverse/blog-app-3/spec/helpers/posts_helper_spec.rb
     # Not yet implemented
     # ./spec/helpers/posts_helper_spec.rb:14

  2) UsersHelper add some examples to (or delete) /home/franclobo/microverse/blog-app-3/spec/helpers/users_helper_spec.rb
     # Not yet implemented
     # ./spec/helpers/users_helper_spec.rb:14


Failures:

  1) Post returns the last five comments on the post
     Failure/Error: expect(subject.recent_comments.length).to eq(5)
     
       expected: 5
            got: 0
     
       (compared using ==)
     # ./spec/models/post_spec.rb:46:in `block (2 levels) in <main>'

  2) PostsController GET /index gets the correct text placeholder from the body
     Failure/Error: expect(response.body).to include('Post')
     
       expected "<!DOCTYPE html>\n<html>\n  <head>\n    <title>BlogApp</title>\n    <meta name=\"viewport\" content=\...ion f\">\n    <a href=\"/users/1/posts/new\">New post</a>\n  </button>\n<div>\n  </body>\n</html>\n" to include "Post"
       Diff:
       @@ -1,52 +1,103 @@
       -Post
       +<!DOCTYPE html>
       +<html>
       +  <head>
       +    <title>BlogApp</title>
       +    <meta name="viewport" content="width=device-width,initial-scale=1">
       +    
       +    
       +
       +    <link rel="stylesheet" href="/assets/application-17de878b28f95a280f7bf4c223707c7c270d3ce052eb3200f5a28fdd76e179a2.css" data-turbo-track="reload" />
       +    <script type="importmap" data-turbo-track="reload">{
       +  "imports": {
       +    "application": "/assets/application-37f365cbecf1fa2810a8303f4b6571676fa1f9c56c248528bc14ddb857531b95.js",
       +    "@hotwired/turbo-rails": "/assets/turbo.min-f309baafa3ae5ad6ccee3e7362118b87678d792db8e8ab466c4fa284dd3a4700.js",
       +    "@hotwired/stimulus": "/assets/stimulus.min-d03cf1dff41d6c5698ec2c5d6a501615a7a33754dbeef8d1edd31c928d17c652.js",
       +    "@hotwired/stimulus-loading": "/assets/stimulus-loading-1fc59770fb1654500044afd3f5f6d7d00800e5be36746d55b94a2963a7a228aa.js",
       +    "controllers/application": "/assets/controllers/application-368d98631bccbf2349e0d4f8269afb3fe9625118341966de054759d96ea86c7e.js",
       +    "controllers/hello_controller": "/assets/controllers/hello_controller-549135e8e7c683a538c3d6d517339ba470fcfb79d62f738a0a089ba41851a554.js",
       +    "controllers": "/assets/controllers/index-2db729dddcc5b979110e98de4b6720f83f91a123172e87281d5a58410fc43806.js"
       +  }
       +}</script>
       +<link rel="modulepreload" href="/assets/application-37f365cbecf1fa2810a8303f4b6571676fa1f9c56c248528bc14ddb857531b95.js">
       +<link rel="modulepreload" href="/assets/turbo.min-f309baafa3ae5ad6ccee3e7362118b87678d792db8e8ab466c4fa284dd3a4700.js">
       +<link rel="modulepreload" href="/assets/stimulus.min-d03cf1dff41d6c5698ec2c5d6a501615a7a33754dbeef8d1edd31c928d17c652.js">
       +<link rel="modulepreload" href="/assets/stimulus-loading-1fc59770fb1654500044afd3f5f6d7d00800e5be36746d55b94a2963a7a228aa.js">
       +<script src="/assets/es-module-shims.min-4ca9b3dd5e434131e3bb4b0c1d7dff3bfd4035672a5086deec6f73979a49be73.js" async="async" data-turbo-track="reload"></script>
       +<script type="module">import "application"</script>
       +  </head>
       +
       +  <body>
       +    <div class="posts-index f c">
       +  <div class="user-info f r">
       +  <img class="user-photo" src="" alt="User Photo">
       +  <div class="user-details f r">
       +    <h3 class="user-name">Eduardo</h3>
       +    <span class="posts-counter">Number of posts: 1</span>
       +  </div>
       +</div>
       +    <a class="post-container f c" href="/users/1/posts/647"> 
       +      <article class="post-info f c">
       +  <h3 class="post-title">first post</h3>
       +  <p class="post-text"></p>
       +  <span class="post-counters">Comments: 0, Likes: 0</span>
       +</article>
       +      
       +    </a>
       +  
       +  <button class="button-a pagination f">
       +    <a href="/users/1/posts/new">New post</a>
       +  </button>
       +<div>
       +  </body>
       +</html>
       
     # ./spec/requests/posts_controller_spec.rb:19:in `block (3 levels) in <main>'

  3) UsersController GET /index gets the correct text placeholder from the body
     Failure/Error: expect(response.body).to include('User Name')
     
       expected "<!DOCTYPE html>\n<html>\n  <head>\n    <title>BlogApp</title>\n    <meta name=\"viewport\" content=\...osts: 0</span>\n  </div>\n</div>\n        </a>\n      </li>\n  </ul>\n<div>\n\n  </body>\n</html>\n" to include "User Name"
       Diff:
       @@ -1,201 +1,401 @@
       -User Name
       +<!DOCTYPE html>
       +<html>
       +  <head>
       +    <title>BlogApp</title>
       +    <meta name="viewport" content="width=device-width,initial-scale=1">
       +    
       +    
       +
       +    <link rel="stylesheet" href="/assets/application-17de878b28f95a280f7bf4c223707c7c270d3ce052eb3200f5a28fdd76e179a2.css" data-turbo-track="reload" />
       +    <script type="importmap" data-turbo-track="reload">{
       +  "imports": {
       +    "application": "/assets/application-37f365cbecf1fa2810a8303f4b6571676fa1f9c56c248528bc14ddb857531b95.js",
       +    "@hotwired/turbo-rails": "/assets/turbo.min-f309baafa3ae5ad6ccee3e7362118b87678d792db8e8ab466c4fa284dd3a4700.js",
       +    "@hotwired/stimulus": "/assets/stimulus.min-d03cf1dff41d6c5698ec2c5d6a501615a7a33754dbeef8d1edd31c928d17c652.js",
       +    "@hotwired/stimulus-loading": "/assets/stimulus-loading-1fc59770fb1654500044afd3f5f6d7d00800e5be36746d55b94a2963a7a228aa.js",
       +    "controllers/application": "/assets/controllers/application-368d98631bccbf2349e0d4f8269afb3fe9625118341966de054759d96ea86c7e.js",
       +    "controllers/hello_controller": "/assets/controllers/hello_controller-549135e8e7c683a538c3d6d517339ba470fcfb79d62f738a0a089ba41851a554.js",
       +    "controllers": "/assets/controllers/index-2db729dddcc5b979110e98de4b6720f83f91a123172e87281d5a58410fc43806.js"
       +  }
       +}</script>
       +<link rel="modulepreload" href="/assets/application-37f365cbecf1fa2810a8303f4b6571676fa1f9c56c248528bc14ddb857531b95.js">
       +<link rel="modulepreload" href="/assets/turbo.min-f309baafa3ae5ad6ccee3e7362118b87678d792db8e8ab466c4fa284dd3a4700.js">
       +<link rel="modulepreload" href="/assets/stimulus.min-d03cf1dff41d6c5698ec2c5d6a501615a7a33754dbeef8d1edd31c928d17c652.js">
       +<link rel="modulepreload" href="/assets/stimulus-loading-1fc59770fb1654500044afd3f5f6d7d00800e5be36746d55b94a2963a7a228aa.js">
       +<script src="/assets/es-module-shims.min-4ca9b3dd5e434131e3bb4b0c1d7dff3bfd4035672a5086deec6f73979a49be73.js" async="async" data-turbo-track="reload"></script>
       +<script type="module">import "application"</script>
       +  </head>
       +
       +  <body>
       +    <div class = "users-container">
       +  <ul class = "users-list f c">
       +      <li>
       +        <a href="/users/1">
       +          <div class="user-info f r">
       +  <img class="user-photo" src="" alt="User Photo">
       +  <div class="user-details f r">
       +    <h3 class="user-name">Eduardo</h3>
       +    <span class="posts-counter">Number of posts: 0</span>
       +  </div>
       +</div>
       +        </a>
       +      </li>
       +      <li>
       +        <a href="/users/2">
       +          <div class="user-info f r">
       +  <img class="user-photo" src="" alt="User Photo">
       +  <div class="user-details f r">
       +    <h3 class="user-name">Eduardo</h3>
       +    <span class="posts-counter">Number of posts: 0</span>
       +  </div>
       +</div>
       +        </a>
       +      </li>
       +      <li>
       +        <a href="/users/3">
       +          <div class="user-info f r">
       +  <img class="user-photo" src="" alt="User Photo">
       +  <div class="user-details f r">
       +    <h3 class="user-name">Eduardo</h3>
       +    <span class="posts-counter">Number of posts: 0</span>
       +  </div>
       +</div>
       +        </a>
       +      </li>
       +      <li>
       +        <a href="/users/45">
       +          <div class="user-info f r">
       +  <img class="user-photo" src="" alt="User Photo">
       +  <div class="user-details f r">
       +    <h3 class="user-name">Eduardo</h3>
       +    <span class="posts-counter">Number of posts: 0</span>
       +  </div>
       +</div>
       +        </a>
       +      </li>
       +      <li>
       +        <a href="/users/46">
       +          <div class="user-info f r">
       +  <img class="user-photo" src="" alt="User Photo">
       +  <div class="user-details f r">
       +    <h3 class="user-name">Eduardo</h3>
       +    <span class="posts-counter">Number of posts: 0</span>
       +  </div>
       +</div>
       +        </a>
       +      </li>
       +      <li>
       +        <a href="/users/47">
       +          <div class="user-info f r">
       +  <img class="user-photo" src="" alt="User Photo">
       +  <div class="user-details f r">
       +    <h3 class="user-name">Eduardo</h3>
       +    <span class="posts-counter">Number of posts: 0</span>
       +  </div>
       +</div>
       +        </a>
       +      </li>
       +      <li>
       +        <a href="/users/89">
       +          <div class="user-info f r">
       +  <img class="user-photo" src="" alt="User Photo">
       +  <div class="user-details f r">
       +    <h3 class="user-name">Eduardo</h3>
       +    <span class="posts-counter">Number of posts: 0</span>
       +  </div>
       +</div>
       +        </a>
       +      </li>
       +      <li>
       +        <a href="/users/90">
       +          <div class="user-info f r">
       +  <img class="user-photo" src="" alt="User Photo">
       +  <div class="user-details f r">
       +    <h3 class="user-name">Eduardo</h3>
       +    <span class="posts-counter">Number of posts: 0</span>
       +  </div>
       +</div>
       +        </a>
       +      </li>
       +      <li>
       +        <a href="/users/91">
       +          <div class="user-info f r">
       +  <img class="user-photo" src="" alt="User Photo">
       +  <div class="user-details f r">
       +    <h3 class="user-name">Eduardo</h3>
       +    <span class="posts-counter">Number of posts: 0</span>
       +  </div>
       +</div>
       +        </a>
       +      </li>
       +      <li>
       +        <a href="/users/133">
       +          <div class="user-info f r">
       +  <img class="user-photo" src="" alt="User Photo">
       +  <div class="user-details f r">
       +    <h3 class="user-name">Eduardo</h3>
       +    <span class="posts-counter">Number of posts: 0</span>
       +  </div>
       +</div>
       +        </a>
       +      </li>
       +      <li>
       +        <a href="/users/134">
       +          <div class="user-info f r">
       +  <img class="user-photo" src="" alt="User Photo">
       +  <div class="user-details f r">
       +    <h3 class="user-name">Eduardo</h3>
       +    <span class="posts-counter">Number of posts: 0</span>
       +  </div>
       +</div>
       +        </a>
       +      </li>
       +      <li>
       +        <a href="/users/135">
       +          <div class="user-info f r">
       +  <img class="user-photo" src="" alt="User Photo">
       +  <div class="user-details f r">
       +    <h3 class="user-name">Eduardo</h3>
       +    <span class="posts-counter">Number of posts: 0</span>
       +  </div>
       +</div>
       +        </a>
       +      </li>
       +      <li>
       +        <a href="/users/177">
       +          <div class="user-info f r">
       +  <img class="user-photo" src="" alt="User Photo">
       +  <div class="user-details f r">
       +    <h3 class="user-name">Eduardo</h3>
       +    <span class="posts-counter">Number of posts: 0</span>
       +  </div>
       +</div>
       +        </a>
       +      </li>
       +      <li>
       +        <a href="/users/178">
       +          <div class="user-info f r">
       +  <img class="user-photo" src="" alt="User Photo">
       +  <div class="user-details f r">
       +    <h3 class="user-name">Eduardo</h3>
       +    <span class="posts-counter">Number of posts: 0</span>
       +  </div>
       +</div>
       +        </a>
       +      </li>
       +      <li>
       +        <a href="/users/179">
       +          <div class="user-info f r">
       +  <img class="user-photo" src="" alt="User Photo">
       +  <div class="user-details f r">
       +    <h3 class="user-name">Eduardo</h3>
       +    <span class="posts-counter">Number of posts: 0</span>
       +  </div>
       +</div>
       +        </a>
       +      </li>
       +  </ul>
       +<div>
       +
       +  </body>
       +</html>
       
     # ./spec/requests/users_controller_spec.rb:17:in `block (3 levels) in <main>'

Finished in 14.24 seconds (files took 2.02 seconds to load)
60 examples, 3 failures, 2 pending

Failed examples:

rspec ./spec/models/post_spec.rb:39 # Post returns the last five comments on the post
rspec ./spec/requests/posts_controller_spec.rb:18 # PostsController GET /index gets the correct text placeholder from the body
rspec ./spec/requests/users_controller_spec.rb:16 # UsersController GET /index gets the correct text placeholder from the body
```

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi Team, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  All required changes implemented
:heavy_check_mark: All tests are passing :+1: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**